include/cuco/priority_queue.cuh

```diff
@@ -0,0 +1,233 @@
+#pragma once
+
+#include <vector>
+#include <utility>
+#include <cuco/detail/pq_pair.cuh>
+
+namespace cuco {
+
+/*
+* @brief A GPU-accelerated priority queue of key-value pairs
```

Comment:
Is there a reason for this to be hardcoded for key-value pairs? Can't it be for any trivially copyable type `T`? e.g., with `std::priority_queue` I could have a `std::priority_queue<int>` or a `std::priority_queue<std::pair<int,int>>`.

Reply:
Update docs now that this has been updated.